tests/test_indexer.py

```diff
@@ -1439,7 +1462,15 @@ def create_index_data(blobstore, bucket_name, bundle_key, manifest,
             except Exception:
                 continue
             index_filename = file_info["name"].replace(".", "_")
-            index_files[index_filename] = file_json
+            if index_filename.endswith('_json'):
```

Comment:
Suggested: This looks like it duplicates functionality from above. Consider extracting a function.

Reply:
This is a small part of a function that reproduces a large chunk of application logic.

Reply:
Requested: I may be mistaken, but these twelve lines are identical to the ones in `BundleDocument.from_bundle()`. Why do we need to duplicate them?